app/cli/cmd/root.go

```diff
@@ -239,46 +268,125 @@ func loadControlplaneAuthToken(cmd *cobra.Command) (string, error) {
 // 3. API token
 // Each one of them have an associated audience claim that we use to identify the type of token. If the token is not
 // present, nor we cannot match it with one of the expected audience, return nil.
-func parseToken(token string) (*ParsedToken, error) {
+func parseToken(token string) (*parsedToken, error) {
```

Comment:
Not for this PR, but most probably we can have a common interface. Something like 
```
type Token interface {
  func Parse(token string) (*parsedToken, error)
}
```
And then moving the "extracting" logic to their own implementations.